terraform/modules/hub/files/cloud-init/prometheus.sh

```diff
@@ -131,10 +131,24 @@ Environment=HTTP_PROXY=${egress_proxy_url_with_protocol}
 Environment=HTTPS_PROXY=${egress_proxy_url_with_protocol}
 EOF
 
-cat <<EOF > /etc/prometheus/prometheus.yml
-${prometheus_config}
-EOF
-
 systemctl daemon-reload
 systemctl enable  prometheus
 systemctl restart prometheus
+
+(crontab -l ;\
+cat <<EOF
+* * * * * \
+  curl ${cronitor_prometheus_config_url}/run -m 10 \
+  && \
+  aws s3 cp \
+    ${config_bucket}/prometheus/prometheus.yml \
+    /tmp/prometheus.yml \
+  && \
+  if !cmp -s /tmp/prometheus.yml /etc/prometheus/prometheus.yml; then \
+    mv /tmp/prometheus.yml /etc/prometheus/prometheus.yml \
+    && systemctl reload prometheus; \
+  fi \
+  && \
+  curl ${cronitor_prometheus_config_url}/complete -m 10 \
```

Comment:
That's `/complete` for success - how do I check whether the cron job failed at all (there's a cronitor `/fail` endpoint that I assume I have to use to report a failure)?

Reply:
I think you should refactor this into a script in `/usr/local/bin` and use `trap` for fail

```
#!/usr/bin/env bash
set -ueo pipefail

: "${CRONITOR_URL:?CRONITOR_URL not set}"

function err {
  curl -sf -m 10 $CRONITOR_URL/fail
}

trap cleanup ERR

curl -sf -m 10 $CRONITOR_URL/run

# do stuff

curl -sf -m 10 $CRONITOR_URL/complete
```

or similar

Reply:
I made a [new branch](https://github.com/alphagov/verify-infrastructure/tree/prometheus-cronitor-script) for this to avoid pushing to this one any more until it's OK, as it's noisy enough already.

Reply:
I think it is done. 🤞I cherry-picked my changes from the other branch.